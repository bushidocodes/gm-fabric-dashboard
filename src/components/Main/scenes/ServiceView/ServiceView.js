import React, { Component } from "react";
import { PropTypes } from "prop-types";
import { Actions } from "jumpstate";
import _ from "lodash";

import Table from "components/Main/components/Table";
import TableToolbar from "components/Main/components/TableToolbar";
import ErrorBoundary from "components/ErrorBoundary";
import NotFoundError from "components/Main/components/NotFoundError";
import { reportError } from "services/notification";
import {
  routerHistoryShape,
  routerLocationShape,
  serviceInstanceShape
} from "components/PropTypes";
import withUrlState from "components/withUrlState";

// TODO: Make history and location a shape PropType
class ServiceView extends Component {
  static propTypes = {
    history: routerHistoryShape.isRequired,
    instances: PropTypes.arrayOf(serviceInstanceShape).isRequired,
    location: routerLocationShape.isRequired,
    serviceIsMetered: PropTypes.bool,
    serviceName: PropTypes.string.isRequired,
    serviceVersion: PropTypes.string.isRequired,
    setUrlState: PropTypes.func.isRequired,
    status: PropTypes.string.isRequired,
    urlState: PropTypes.object.isRequired
  };

  componentDidMount() {
    const { location: { state }, history } = this.props;
    if (state && state.message) {
      // Disable polling and clear metrics cache
      Actions.stopPollingAndPurgeInstanceMetrics();
      // Display notification
      reportError(state.message);
      // Reset location state
      history.replace({
        state: {}
      });
    }
  }

  setSortByAttribute = newSortByAttribute => {
    const {
      urlState: { ascending = "true", sortByAttribute = "name" },
      setUrlState
    } = this.props;
    if (sortByAttribute === newSortByAttribute) {
      setUrlState({ ascending: !JSON.parse(ascending) });
    } else {
      setUrlState({ sortByAttribute: newSortByAttribute });
    }
  };

  render() {
    const {
      setUrlState,
      serviceName,
      serviceVersion,
      status,
      instances,
      serviceIsMetered,
      urlState: {
        filterString = "",
        sortByAttribute = "name",
        ascending = "true"
      }
    } = this.props;

    const sortOrder = JSON.parse(ascending) ? ["asc"] : ["desc"];

    return instances && instances.length ? (
      <div>
        <TableToolbar
          searchInputProps={{
            filterString,
            setFilterString: filterString => setUrlState({ filterString }),
            searchPlaceholder: "Search Instances"
          }}
          sortByProps={{
            sortByAttribute,
            sortByOptions: [
              {
                value: "name",
                label: "Name"
              },
              {
                value: "start_time",
                label: "Uptime"
              }
            ],
            setSortByAttribute: this.setSortByAttribute
          }}
        />
        <ErrorBoundary>
          <Table
            type={"Instance"}
            serviceName={serviceName}
            serviceVersion={serviceVersion}
            serviceIsMetered={serviceIsMetered}
            items={_.orderBy(
              instances.filter(
                ({ name }) =>
                  name.toLowerCase().indexOf(filterString.toLowerCase()) !== -1
              ),
              [sortByAttribute.toLowerCase()],
              sortOrder
            )}
            status={status}
          />
        </ErrorBoundary>
      </div>
    ) : (
      <NotFoundError errorMsg={"No Instances Found"} />
    );
  }
}

export default withUrlState()(ServiceView);
