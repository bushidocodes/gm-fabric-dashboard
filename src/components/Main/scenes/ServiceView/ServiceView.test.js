import React from "react";
import { mount } from "enzyme";
import { MemoryRouter, Route } from "react-router";
import renderer from "react-test-renderer";

import ServiceView from "./ServiceView";

const mockData = {
  serviceName: "Cool Service",
  serviceVersion: "1.0",
  status: "stable",
  instances: [
    { name: "serviceID1", start_time: 600000000000 },
    { name: "serviceID2", start_time: 300000000000 },
    { name: "serviceID3", start_time: 400000000000 }
  ]
};

// Wrap Service View in Memory Router to mock route props (history, match, location)
const RouterWrap = (
  <MemoryRouter>
    <Route render={props => <ServiceView {...props} {...mockData} />} />
  </MemoryRouter>
);

const tableProps = {
  serviceName: "Cool Service",
  serviceVersion: "1.0",
  status: "stable",
  type: "Instance",
  items: [
    { name: "serviceID1", start_time: 600000000000 },
    { name: "serviceID2", start_time: 300000000000 },
    { name: "serviceID3", start_time: 400000000000 }
  ]
};

describe("Service View", () => {
  let wrapper = mount(RouterWrap);

  test("matches snapshot", () => {
    const tree = renderer.create(RouterWrap).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders all instances", () => {
    expect(wrapper.find("GMServiceTableLineItem").length).toBe(3);
    expect(wrapper.html().includes("serviceID1")).toBe(true);
    expect(wrapper.html().includes("serviceID2")).toBe(true);
    expect(wrapper.html().includes("serviceID3")).toBe(true);
  });

  test("passes the correct props down to TableToolbar", () => {
    expect(wrapper.find("TableToolbar").props()).toMatchObject({
      searchInputProps: {
        filterString: "",
        searchPlaceholder: "Search Instances"
      },
      sortByProps: {
        sortByAttribute: "name",
        sortByOptions: [
          { label: "Name", value: "name" },
          { label: "Uptime", value: "start_time" }
        ]
      }
    });
  });

  test("passes the correct props down to ThreadsTable", () => {
    expect(wrapper.find("Table").props()).toMatchObject(tableProps);
  });
});
