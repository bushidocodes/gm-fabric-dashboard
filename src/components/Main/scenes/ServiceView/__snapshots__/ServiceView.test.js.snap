// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Service View matches snapshot 1`] = `
.c14 {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c13 {
  z-index: 1;
  height: 24px;
  width: 24px;
  color: currentColor;
  word-spacing: 0;
  vertical-align: middle;
}

.c13:not(:root) {
  overflow: visible;
}

.c11 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgb(255,255,255);
  box-shadow: inset 0 -1px hsl(0,0%,92%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  min-height: 36px;
  width: 100%;
  cursor: default;
}

.c11:focus {
  outline: -webkit-focus-ring-color auto 5px;
}

.c11:focus:active:hover {
  outline: 0;
}

.c11 > * {
  padding-right: 16px;
  padding-top: 7px;
  padding-bottom: 7px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: visible;
}

.c11 > *:first-child {
  padding-left: 16px;
}

.c12 {
  -webkit-flex: 1 1 12%;
  -ms-flex: 1 1 12%;
  flex: 1 1 12%;
  text-align: left;
}

.c18 {
  -webkit-flex: 1 1 12%;
  -ms-flex: 1 1 12%;
  flex: 1 1 12%;
  text-align: left;
  text-align: right;
}

.c18 span + span {
  font-size: 80%;
  opacity: 0.8;
}

.c16 {
  width: 100%;
  cursor: pointer;
  text-decoration: none;
  color: black;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: rgb(77,77,77);
  cursor: not-allowed;
}

.c16:focus {
  outline: -webkit-focus-ring-color auto 5px;
}

.c16:hover {
  color: rgb(77,77,77);
  cursor: not-allowed;
}

.c16:focus:active:hover {
  outline: 0;
}

.c17 {
  background-color: rgba(0,0,0,0.8);
  border-radius: 6px;
  color: rgb(255,255,255);
  font-family: Metropolis,-apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen","Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue",sans-serif;
  font-size: 11px;
  opacity: 0;
  padding: 10px;
  position: absolute;
  text-align: left;
  -webkit-transition: opacity 1s;
  transition: opacity 1s;
  visibility: hidden;
  white-space: normal;
  width: 160px;
  z-index: 1070;
  right: 120%;
  top: -5px;
}

.c15 {
  border-bottom: 1px solid rgb(217,217,217);
  cursor: help;
  display: inline-block;
  position: relative;
}

.c15:hover > :last-child {
  opacity: 1;
  visibility: visible;
  z-index: 1070;
}

.c19 span + span {
  font-size: 80%;
  opacity: 0.8;
}

.c7 {
  display: block;
  font-family: Rubik,-apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen","Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue",sans-serif;
  margin-left: auto;
  margin-right: auto;
  max-width: 1300px;
  width: 100%;
}

.c8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgb(255,255,255);
  box-shadow: inset 0 -1px hsl(0,0%,92%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  font-weight: 400;
  overflow: visible;
  width: 100%;
}

.c10 {
  display: block;
  list-style: none;
  margin: 0;
  padding: 0;
}

.c9 {
  -webkit-flex: 1 1 12%;
  -ms-flex: 1 1 12%;
  flex: 1 1 12%;
  text-align: left;
  padding-right: 16px;
  padding-top: 7px;
  padding-bottom: 7px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c9:first-child {
  padding-left: 16px;
}

.c2 {
  background-image: url(magnifier.svg);
  background-repeat: no-repeat;
  background-size: contain;
  position: relative;
  text-indent: 20px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: black;
  font-family: Metropolis,-apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen","Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue",sans-serif;
  font-weight: 600;
  font-size: 14px;
  line-height: 1.4;
  text-align: left;
  text-transform: none;
  padding: 1.4px 4px;
  background-color: rgb(255,255,255);
  border-radius: 3px;
  border: 1px solid rgb(242,242,242);
  border-bottom-color: rgb(230,230,230);
  -webkit-user-select: auto;
  -moz-user-select: auto;
  -ms-user-select: auto;
  user-select: auto;
  box-shadow: inset 0 0 0 transparent,0 0 0 transparent,0 0 0 transparent;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
}

.c2::-webkit-search-cancel-button {
  cursor: pointer;
}

.c2:hover {
  background-color: rgb(250,250,250);
  border-color: rgb(237,237,237);
  -webkit-transition: all 0.1s ease;
  transition: all 0.1s ease;
  border-bottom-color: rgb(217,217,217);
  color: black;
}

.c2:focus {
  outline: none;
  box-shadow: 0 0 0 2px rgb(10,171,42);
  z-index: 1;
}

.c2::-webkit-input-placeholder {
  color: currentColor;
  opacity: 1;
  text-align: center;
}

.c2::-moz-placeholder {
  color: currentColor;
  opacity: 1;
  text-align: center;
}

.c2:-ms-input-placeholder {
  color: currentColor;
  opacity: 1;
  text-align: center;
}

.c2::placeholder {
  color: currentColor;
  opacity: 1;
  text-align: center;
}

.c2::-ms-expand {
  background-color: transparent;
  border: 0;
}

.c5 {
  font-family: Metropolis,-apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen","Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue",sans-serif;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  font-size: 14px;
  font-weight: 600;
  height: 28px;
  max-width: 125px;
  position: relative;
  width: 100%;
  z-index: 1020;
}

.c5 .Select-control {
  height: 28px !important;
}

.c5 .Select-control .Select-placeholder {
  line-height: 28px;
}

.c5 .Select-control .Select-value {
  line-height: 28px !important;
}

.c5 .Select-control .Select-value .Select-value-label {
  line-height: 28px;
}

.c5 .Select-control .Select-input {
  height: 28px;
}

.c6 {
  margin-left: 4px;
  opacity: 0.6;
  font-size: 11px;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: hsl(0,0%,96%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 36px;
  margin-left: auto;
  margin-right: auto;
  width: 100%;
  max-width: 1300px;
  padding: 2px 16px;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 33%;
  -ms-flex: 1 1 33%;
  flex: 1 1 33%;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 33%;
  -ms-flex: 1 1 33%;
  flex: 1 1 33%;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 33%;
  -ms-flex: 1 1 33%;
  flex: 1 1 33%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media all and (min-width:1300px) {
  .c0:after {
    background-color: inherit;
    content: "";
    height: inherit;
    left: 0;
    pointer-events: none;
    position: absolute;
    right: 0;
    top: 0;
    z-index: 1;
  }

  .c0 > * {
    position: relative;
    z-index: 2;
  }
}

<div>
  <div
    className="c0"
  >
    <div
      className="c1"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          aria-label="Search Instances"
          className="c2"
          onChange={[Function]}
          placeholder="Search Instances"
          type="search"
          value=""
        />
      </form>
    </div>
    <div
      className="c3"
    />
    <div
      className="c4"
    >
      <div
        className="Select c5 has-value is-searchable Select--single"
        style={undefined}
      >
        <input
          disabled={false}
          name="form-field-sort-by"
          type="hidden"
          value="name"
        />
        <div
          className="Select-control"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={undefined}
        >
          <span
            className="Select-multi-value-wrapper"
            id="react-select-3--value"
          >
            <div
              className="Select-value"
              style={undefined}
              title={undefined}
            >
              <span
                aria-selected="true"
                className="Select-value-label"
                id="react-select-3--value-item"
                role="option"
              >
                <span>
                  <span>
                    Sort
                     
                  </span>
                  <span
                    className="c6"
                  >
                    Name
                  </span>
                </span>
              </span>
            </div>
            <div
              className="Select-input"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-activedescendant="react-select-3--value"
                aria-describedby={undefined}
                aria-expanded="false"
                aria-haspopup="false"
                aria-label={undefined}
                aria-labelledby={undefined}
                aria-owns=""
                className={undefined}
                id={undefined}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                role="combobox"
                style={
                  Object {
                    "boxSizing": "content-box",
                    "width": "5px",
                  }
                }
                tabIndex={undefined}
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </span>
          <span
            className="Select-arrow-zone"
            onMouseDown={[Function]}
          >
            <span
              className="Select-arrow"
              onMouseDown={[Function]}
            />
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="c7"
  >
    <div
      className="c8"
    >
      <div
        className="c9"
        style={
          Object {
            "flex": "1 1 20%",
          }
        }
      >
        Instance
      </div>
      <div
        className="c9"
        style={
          Object {
            "textAlign": "right",
          }
        }
      >
        Uptime
      </div>
    </div>
    <ol
      className="c10"
    >
      <li
        className="c11"
        onClick={[Function]}
        tabIndex="-1"
      >
        <div
          className="c12"
          style={
            Object {
              "flex": "1 1 20%",
            }
          }
        >
          <svg
            aria-labelledby="ariaLabelledby"
            className="c13"
            focusable="false"
            version="1.1"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <g
              className="c14"
              fill="currentColor"
              title=""
            >
              <title>
                No Metrics
              </title>
              <path
                d="M11.5765139,10.7780593 L12.5284275,6.56244205 C12.6888683,5.85191831 13.701872,5.85364698 13.859887,6.56471416 L15.1056286,12.1705517 L18.3106287,12.5116668 L18.9928589,12.8527819 C18.5380388,13.0801919 18.3106287,13.1938969 18.3106287,13.1938969 C18.3106287,13.1938969 17.0598735,13.3076019 14.558363,13.535012 C14.4557501,13.535012 14.3575553,13.5121413 14.2692177,13.4707631 L17.8355683,17.0371136 L16.9782404,17.8944415 L6.42135469,7.33755581 L7.27868258,6.48022793 L11.5765139,10.7780593 Z M13.9403445,13.1418898 C13.9195802,13.0974656 13.9033732,13.0502532 13.8923788,13.0007784 L13.1885769,9.83366984 L12.7176514,11.9191968 L13.9403438,13.1418892 Z M8.43252693,10.9345145 L10.2560316,12.7580191 L10.7377973,14.4923757 L10.9685377,13.4705253 L12.1096752,14.6116628 L11.4715725,17.4375463 C11.3153851,18.1292335 10.3385431,18.1531049 10.1487563,17.4698724 L8.86830577,12.8602505 L8.66218424,13.2037864 C8.5388894,13.4092778 8.31681883,13.535012 8.07717673,13.535012 L5.68937126,13.1938969 L5.00714113,12.8527819 C5.46196122,12.6253718 5.68937126,12.5116668 5.68937126,12.5116668 C5.68937126,12.5116668 6.35654904,12.3979618 7.69090459,12.1705517 L8.43252693,10.9345145 Z"
                id="No-Metrics"
              />
            </g>
          </svg>
          <div
            className="c15"
            disabled={undefined}
            style={
              Object {
                "border": "none",
              }
            }
          >
            <a
              className="c16"
              disabled={true}
              href="/Cool·Service/1.0/serviceID1"
              onClick={[Function]}
            >
              serviceID1
            </a>
            <div
              className="c17"
              style={Object {}}
            >
              This microservice instance does not have metrics to display.
            </div>
          </div>
        </div>
        <div
          className="c18"
        >
          <div
            className="c19"
          />
        </div>
      </li>
      <li
        className="c11"
        onClick={[Function]}
        tabIndex="-1"
      >
        <div
          className="c12"
          style={
            Object {
              "flex": "1 1 20%",
            }
          }
        >
          <svg
            aria-labelledby="ariaLabelledby"
            className="c13"
            focusable="false"
            version="1.1"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <g
              className="c14"
              fill="currentColor"
              title=""
            >
              <title>
                No Metrics
              </title>
              <path
                d="M11.5765139,10.7780593 L12.5284275,6.56244205 C12.6888683,5.85191831 13.701872,5.85364698 13.859887,6.56471416 L15.1056286,12.1705517 L18.3106287,12.5116668 L18.9928589,12.8527819 C18.5380388,13.0801919 18.3106287,13.1938969 18.3106287,13.1938969 C18.3106287,13.1938969 17.0598735,13.3076019 14.558363,13.535012 C14.4557501,13.535012 14.3575553,13.5121413 14.2692177,13.4707631 L17.8355683,17.0371136 L16.9782404,17.8944415 L6.42135469,7.33755581 L7.27868258,6.48022793 L11.5765139,10.7780593 Z M13.9403445,13.1418898 C13.9195802,13.0974656 13.9033732,13.0502532 13.8923788,13.0007784 L13.1885769,9.83366984 L12.7176514,11.9191968 L13.9403438,13.1418892 Z M8.43252693,10.9345145 L10.2560316,12.7580191 L10.7377973,14.4923757 L10.9685377,13.4705253 L12.1096752,14.6116628 L11.4715725,17.4375463 C11.3153851,18.1292335 10.3385431,18.1531049 10.1487563,17.4698724 L8.86830577,12.8602505 L8.66218424,13.2037864 C8.5388894,13.4092778 8.31681883,13.535012 8.07717673,13.535012 L5.68937126,13.1938969 L5.00714113,12.8527819 C5.46196122,12.6253718 5.68937126,12.5116668 5.68937126,12.5116668 C5.68937126,12.5116668 6.35654904,12.3979618 7.69090459,12.1705517 L8.43252693,10.9345145 Z"
                id="No-Metrics"
              />
            </g>
          </svg>
          <div
            className="c15"
            disabled={undefined}
            style={
              Object {
                "border": "none",
              }
            }
          >
            <a
              className="c16"
              disabled={true}
              href="/Cool·Service/1.0/serviceID2"
              onClick={[Function]}
            >
              serviceID2
            </a>
            <div
              className="c17"
              style={Object {}}
            >
              This microservice instance does not have metrics to display.
            </div>
          </div>
        </div>
        <div
          className="c18"
        >
          <div
            className="c19"
          />
        </div>
      </li>
      <li
        className="c11"
        onClick={[Function]}
        tabIndex="-1"
      >
        <div
          className="c12"
          style={
            Object {
              "flex": "1 1 20%",
            }
          }
        >
          <svg
            aria-labelledby="ariaLabelledby"
            className="c13"
            focusable="false"
            version="1.1"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <g
              className="c14"
              fill="currentColor"
              title=""
            >
              <title>
                No Metrics
              </title>
              <path
                d="M11.5765139,10.7780593 L12.5284275,6.56244205 C12.6888683,5.85191831 13.701872,5.85364698 13.859887,6.56471416 L15.1056286,12.1705517 L18.3106287,12.5116668 L18.9928589,12.8527819 C18.5380388,13.0801919 18.3106287,13.1938969 18.3106287,13.1938969 C18.3106287,13.1938969 17.0598735,13.3076019 14.558363,13.535012 C14.4557501,13.535012 14.3575553,13.5121413 14.2692177,13.4707631 L17.8355683,17.0371136 L16.9782404,17.8944415 L6.42135469,7.33755581 L7.27868258,6.48022793 L11.5765139,10.7780593 Z M13.9403445,13.1418898 C13.9195802,13.0974656 13.9033732,13.0502532 13.8923788,13.0007784 L13.1885769,9.83366984 L12.7176514,11.9191968 L13.9403438,13.1418892 Z M8.43252693,10.9345145 L10.2560316,12.7580191 L10.7377973,14.4923757 L10.9685377,13.4705253 L12.1096752,14.6116628 L11.4715725,17.4375463 C11.3153851,18.1292335 10.3385431,18.1531049 10.1487563,17.4698724 L8.86830577,12.8602505 L8.66218424,13.2037864 C8.5388894,13.4092778 8.31681883,13.535012 8.07717673,13.535012 L5.68937126,13.1938969 L5.00714113,12.8527819 C5.46196122,12.6253718 5.68937126,12.5116668 5.68937126,12.5116668 C5.68937126,12.5116668 6.35654904,12.3979618 7.69090459,12.1705517 L8.43252693,10.9345145 Z"
                id="No-Metrics"
              />
            </g>
          </svg>
          <div
            className="c15"
            disabled={undefined}
            style={
              Object {
                "border": "none",
              }
            }
          >
            <a
              className="c16"
              disabled={true}
              href="/Cool·Service/1.0/serviceID3"
              onClick={[Function]}
            >
              serviceID3
            </a>
            <div
              className="c17"
              style={Object {}}
            >
              This microservice instance does not have metrics to display.
            </div>
          </div>
        </div>
        <div
          className="c18"
        >
          <div
            className="c19"
          />
        </div>
      </li>
    </ol>
  </div>
</div>
`;
